src/item.cpp

```diff
@@ -5006,6 +5007,10 @@ bool item::reduce_charges( long quantity )
     return false;
 }
 
+void item::muzzle_flash_falloff(){
+    light.luminance > type->gun->muzzle_flash / 4 ? light.luminance /= 2 : light.luminance = 0;  
```

Comment:
And it would be nice if you add a check for whether this item actually *is* a gun before you access `type->gun->...`, because if it's not, the game will crash here.

Also have you heard of the new thing, called an "if-statement"? You can use it like this:
```C++
if( light.luminance > type->gun->muzzle_flash / 4 ) {
    light.luminance /= 2;
} else {
    light.luminance = 0;
}
```
Putting this logic into the ternary operator is very obfuscating.